### ISSUE TITLE
Cannot use Microsoft.ML.OnnxRuntime.DirectML and WinUI 3 desktop: The target process exited without raising a CoreCLR started event

### ISSUE THREAD

**Issue by jdluzen** [microsoft/onnxruntime#7237]
**Describe the bug**
I am attempting to build a machine learning application with the recently released WinUI 3 desktop. Adding the package `Microsoft.ML.OnnxRuntime.DirectML` to an out of the box WinUI 3 app results in it being unable to run or debug with the following: `The target process exited without raising a CoreCLR started event. Ensure that the target process is configured to use .NET Core. This may be expected if the target process did not run on .NET Core.`

**Urgency**
While I do not technically have a deadline, it is a complete blocker, and I cannot continue. After searching and debugging for many hours over at least two days, I have found no workarounds so far.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `Windows 10 2004`
- ONNX Runtime installed from (source or binary): `binary Nuget`
- ONNX Runtime version: `1.7.0`
- Python version: N/A
- Visual Studio version (if applicable): `2019 16.9.3` and `2019 16.10.0 Preview 1.0`
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: `5700 XT 8GB`

**To Reproduce**
- Create a new Blank App, Packaged (WinUI 3 in Desktop)
- Add package reference to `Microsoft.ML.OnnxRuntime.DirectML`
- Run (I use x64, but same behavior on x86)

**Expected behavior**
The default WinUI 3 window to appear

**Screenshots**
N/A

**Additional context**
Some searches have suggested that upgrading the dependencies would resolve it, but the same behavior occurs.
Relevant csproj:
```
    <TargetFramework>net5.0-windows10.0.19041.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
```
Though I use the DirectML package, I suspect it's unrelated to it specifically. It works, at least it runs and debugger attaches, with a .net5 console app.

**Comment by fdwr** [comment]
Joe, is this *specific* to `Microsoft.ML.OnnxRuntime.DirectML`, yet works fine with `Microsoft.ML.OnnxRuntime`? (https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime) I'm perplexed because ORT calls DirectML.dll, but DirectML has no C# or .NET dependencies itself. It's actually *ORT* which projects the C# API to the caller, above DML. @smk2007 informs me that `Microsoft.ML.OnnxRuntime.DirectML` depends on `Microsoft.ML.OnnxRuntime.Managed`, which pulls in .NETStandard v1.1.

**Comment by jdluzen** [comment]
You're correct, it does occur with `Microsoft.ML.OnnxRuntime` too, but not `Microsoft.ML.OnnxRuntime.Managed`. I'll edit the original post.

**Comment by fdwr** [comment]
@faxu: Turns out this wasn't DML specific and occurs with the CPU package too. Can someone more familiar with ORT's C# wrapper look?

**Comment by jdluzen** [comment]
I've found a workaround:
do not reference `Microsoft.ML.OnnxRuntime`, but include `onnxruntime.dll` from the Nuget cache in the project, and copy to output.

**Comment by fdwr** [comment]
@jdluzen: I'm really not sure who to assign this issue to on ORT, but I see @jywu-msft reviewing !7540. So I'll try him, and he might know who to look into it further.
